ECMA2Yaml/ECMAHelper/YamlConverters/SDPYamlConverter.Delegate.cs

```diff
@@ -17,14 +17,35 @@ public DelegateSDPModel FormatDelegate(Type t)
             sdpDelegate.TypeParameters = ConvertTypeParameters(t);
             sdpDelegate.Inheritances = t.InheritanceChains?.LastOrDefault().Values.Select(uid => UidToTypeMDString(uid, _store)).ToList();
 
-            if (t.ReturnValueType != null
-                && !string.IsNullOrEmpty(t.ReturnValueType.Type)
-                && t.ReturnValueType.Type != "System.Void"
-                && t.ItemType != ItemType.Event)
+            if (t.ReturnValueType != null && t.ItemType != ItemType.Event)
             {
-                sdpDelegate.Returns = ConvertParameter<TypeReference>(t.ReturnValueType);
-            }
+                var returns = t.ReturnValueType;
 
+                if (returns.VersionedTypes.Count() == 1)
```

Comment:
can these logic be shared between delegate and member converters?

Reply:
I'd considered that even in my first iteration ... but this section of the code effectively has _two_ return types, depending on if the value is monikerized or not (ie. setting .Returns, or ReturnsWithMoniker). So to avoid having a helper method with complicated out/ref semantics or something else, I just effectively copied these two routines in the two places where it was used.